### PR TITLE
chore(infra): add the npm runner token

### PR DESCRIPTION
**Pull request by mr-uniswap**
## Changes
Moves our deploy to the new npm runner (makes it faster and more secure)

#### Before merging, please ensure the following:

- [ ] All events have been approved by both product and engineering
- [ ] All events have been tested in their consuming package
- [ ] This PR is titled as `feat` for any new events, or otherwise follows conventional commit standards

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -8,7 +8,8 @@ on:
 jobs:
   release:
     name: Release
-    runs-on: ubuntu-latest
+    runs-on: 
+      npm-deploy
```
Comment (adjkant) [inline .github/workflows/release.yaml:12]: does this need `group:`?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -29,7 +30,7 @@ jobs:
           export-env: true
         env:
           OP_SERVICE_ACCOUNT_TOKEN: ${{ secrets.OP_SERVICE_ACCOUNT_TOKEN }}
-          NPM_TOKEN: op://npm-deploy/npm-token/secret
+          NPM_TOKEN: op://npm-deploy/npm-runner-token/secret
```
Comment (adjkant) [inline .github/workflows/release.yaml:33]: What is this change for?

Comment (mr-uniswap) [reply .github/workflows/release.yaml:33]: Switches out the token to a longer lived one that only runs in the IP range of the run group

File: .github/workflows/release.yaml (hunk 1 above)
Comment (mr-uniswap) [reply .github/workflows/release.yaml:12]: Yeah! Thank you!

**Review by adjkant — APPROVED** [review]

**Pull request merged**